src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -26,37 +26,93 @@
 #include <multipass/format.h>
 
 #include <iostream>
+#include <sstream>
 
 namespace mp = multipass;
 namespace mpl = multipass::logging;
 
 namespace
 {
 constexpr auto category = "sshfs mount";
-template <typename Callable>
-auto run_cmd(mp::SSHSession& session, std::string&& cmd, Callable&& error_handler)
+const std::string fuse_version_string{"FUSE library version "};
+const std::string ld_library_path_key{"LD_LIBRARY_PATH="};
+const std::string snap_path_key{"SNAP="};
+
+auto run_cmd(mp::SSHSession& session, std::string&& cmd)
 {
     auto ssh_process = session.exec(cmd);
     if (ssh_process.exit_code() != 0)
-        error_handler(ssh_process);
+        throw std::runtime_error(ssh_process.read_std_error());
+
     return ssh_process.read_std_output();
 }
 
-auto run_cmd(mp::SSHSession& session, std::string&& cmd)
+auto match_line(std::istringstream& output, const std::string& matcher)
 {
-    auto error_handler = [](mp::SSHProcess& proc) { throw std::runtime_error(proc.read_std_error()); };
-    return run_cmd(session, std::forward<std::string>(cmd), error_handler);
+    output.seekg(0, output.beg);
+
+    std::string line;
+    while (std::getline(output, line, '\n'))
+    {
+        if (line.find(matcher) != std::string::npos)
+        {
+            return line;
+        }
+    }
+
+    return std::string{};
 }
 
-void check_sshfs_exists(mp::SSHSession& session)
+auto get_sshfs_exec_and_options(mp::SSHSession& session)
 {
-    auto error_handler = [](mp::SSHProcess& proc) {
-        mpl::log(mpl::Level::warning, category,
-                 fmt::format("Unable to determine if 'sshfs' is installed: {}", proc.read_std_error()));
-        throw mp::SSHFSMissingError();
-    };
-
-    run_cmd(session, "which sshfs", error_handler);
+    std::string sshfs_exec;
+
+    try
+    {
+        // Prefer to use Snap package version first
+        std::istringstream sshfs_env{run_cmd(session, "sudo multipass-sshfs.env")};
+
+        auto ld_library_path = match_line(sshfs_env, ld_library_path_key);
+        auto snap_path = match_line(sshfs_env, snap_path_key);
+        snap_path = snap_path.substr(snap_path_key.length());
+
+        sshfs_exec = fmt::format("{} {}/bin/sshfs", ld_library_path, snap_path);
+    }
+    catch (const std::exception& e)
+    {
+        mpl::log(mpl::Level::debug, category, fmt::format("'sshfs' snap package is not installed: {}", e.what()));
```

Comment:
```suggestion
        mpl::log(mpl::Level::debug, category, fmt::format("'multipass-sshfs' snap package is not installed: {}", e.what()));
```

Or just drop `'sshfs' `, we know where this is.